src/routes/userRoutes.js

```diff
@@ -12,12 +12,14 @@ const {
   updateCurrentUser,
   updateUserPassword,
   deleteSingleUser,
+  getCurrentUserWithPurchasedAlbums,
 } = require('../controllers/userController');
 
 // Define routes for handling user-related operations
 router
   .route('/')
-  .get(authenticateUser, authorizePermissions('admin'), getAllUsers);
+  .get(authenticateUser, authorizePermissions('admin'), getAllUsers)
+  .get(authenticateUser, getCurrentUserWithPurchasedAlbums);
```

Comment:
I think this means both controller handlers are trying to use the path `GET /api/v1/users`.  I think in the end Express will overwrite the first one with the second one.

We probably don't want that so we should have a separate endpoint for getting a user with their albums. We could do something like:

```
router.route('/showMe/withAlbums').get(authenticateUser, getCurrentUserWithPurchasedAlbums)
```

or if we want to keep it as part of the existing `showMe` endpoint and have people do `GET /api/v1/users/showMe?includeAlbums=true`, then we could do:

```
router.route('/showMe').get(authenticateUser, showCurrentUser)

# then inside userController
const showCurrentUser = async (req, res) => {
  if (req.query.includeAlbums === "true") {
    // have the other handler handle this request instead
    return getCurrentUserWithPurchasedAlbums(req, res)
  } else {
    const user = await User.findOne({ _id: req.user.userId }).select('-password');

    if (!user) {
      throw new CustomError.NotFoundError('User not found');
    }

    res.status(StatusCodes.OK).json({ user });
  }
};
```